h5grove/utils.py

```diff
@@ -125,17 +124,29 @@ def _sanitize_dtype(dtype: np.dtype) -> np.dtype:
     return result
 
 
-def sanitize_array(array: Sequence[Number], copy: bool = True) -> np.ndarray:
-    """Ensure array save as .npy can be read back by js-numpy-parser.
+def convert(
+    data: Union[np.ndarray, np.number, np.bool_], dtype: Optional[str] = "origin"
+) -> Union[np.ndarray, np.number, np.bool_]:
```

Comment:
It takes numpy numbers, not only arrays as input to fit in `DatasetContent` context

Reply:
You could have a [TypeVar](https://docs.python.org/3/library/typing.html#typing.TypeVar) here as `np.ndarray` will return `np.ndarray`, `np.number` will return `np.number` and so on.

```python
T = TypeVar('T', np.ndarray, np.number, np.bool_)
def convert(data: T, dtype: Optional[str] = "origin") -> T:
    ...
```